### PR TITLE
Remove duplicated usage of `-Zunstable-options` in bootstrap

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Surprisingly, sometimes it is in the same function.

r? @Kobzol

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Fixed this case. I think a `try` might be welcome too.

@bors try

File: src/bootstrap/src/bin/rustdoc.rs [hunk 1]
```diff
@@ -59,8 +59,6 @@ fn main() {
     if stage == "0" {
         cmd.arg("--cfg=bootstrap");
     }
-    cmd.arg("-Zunstable-options");
-    cmd.arg("--check-cfg=cfg(bootstrap)");
```
Comment (Kobzol) [inline src/bootstrap/src/bin/rustdoc.rs:63]: Why was the `--check-cfg` flag removed?

Comment (GuillaumeGomez) [reply src/bootstrap/src/bin/rustdoc.rs:63]: Duplicated as well.

Comment (Kobzol) [reply src/bootstrap/src/bin/rustdoc.rs:63]: Verified with `--dump-bootstrap-shims`. Thanks!

**Review by Kobzol — APPROVED** [review]
I checked the behavior with `--dump-bootstrap-shims`, looks like this indeed removed a bunch of dupication. This could also help with https://github.com/rust-lang/rust/pull/116278, I remember that there were some issues with some bootstrap invocations missing the unstable options.

It's difficult to expect what could happen in all the possible situations, so let's just check if this passes CI.

**Comment by GuillaumeGomez** [comment]
Didn't know about `--dump-bootstrap-shims`. In my case I looked at rustdoc `--test` arguments haha.

**Pull request merged**